x-pack/plugins/kubernetes_security/public/components/count_widget/hooks.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useInfiniteQuery } from 'react-query';
+import { CoreStart } from '@kbn/core/public';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { QUERY_KEY_COUNT_WIDGET, COUNT_ROUTE } from '../../../common/constants';
+
+export const useFetchCountWidgetData = (
+  widgetKey: string,
+  filterQuery: string,
+  groupedBy: string,
+  index?: string
+) => {
+  const { http } = useKibana<CoreStart>().services;
+  const cachingKeys = [QUERY_KEY_COUNT_WIDGET, widgetKey, filterQuery, groupedBy, index];
```

Comment:
When a user hits 'refresh' at the top level, are we busting this cache somehow? cc @zizhouW @opauloh

Reply:
no, good point! let's handle that apart in a separate PR so we can address all hooks, maybe we can update them to use `lastUpdated` as a `cachingKey `

Reply:
I believe the cache are being invalidated, at least for other chats and tree nav. Just tested it locally - 
Before refresh
<img width="1410" alt="image" src="https://user-images.githubusercontent.com/16872649/178587354-ab35b9c4-1a4f-4ef2-b002-b4747571bfc1.png">

After refresh
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/16872649/178587430-e248bef6-4195-4f9b-a905-73c203ef558f.png">
The 3 aggregates are for the 2 percent charts and the 1 for tree nav

@animehart can you test it for count widgets?

Reply:
<img width="1647" alt="Screen Shot 2022-07-12 at 2 59 17 PM" src="https://user-images.githubusercontent.com/8703149/178611172-10c927e3-07ae-4f3d-9552-b4c886a70bd9.png">
 count route gets called when pressing refresh @opauloh @mitodrummer